cmd/flipt/main.go

```diff
@@ -158,9 +164,25 @@ func main() {
 	}
 }
 
+// determinePath will figure out which path to use for Flipt configuration.
+func determinePath(cfgPath string) string {
+	if cfgPath != "" {
+		return cfgPath
+	}
+
+	_, err := os.Stat(fliptConfigFile)
+	if err == nil {
+		return fliptConfigFile
+	}
+
```

Comment:
The following addition would provide some useful insight when there other kinds of error here (e.g. permission).
This would require making another default logger for `warn` level like with fatal.
Could just make a `defaultLogger` variable instead and all the methods on them instead.
```suggestion
	if !errors.Is(err, fs.ErrNotExist) {
		warn("failed to inspect configuration; falling back to default", zap.String("config_path", path), zap.Error(err))
	}
```